src/features/layers_in_area/atoms/areaLayersLegendsAndSources.ts

```diff
@@ -11,9 +11,12 @@ import type { LayerInAreaDetails } from '../types';
 import type { LayerSource } from '~core/logical_layers/types/source';
 import type { LayerLegend } from '~core/logical_layers/types/legends';
 
-function convertDetailsToSource(response: LayerInAreaDetails): LayerSource {
-  /* Typescript makes me sad sometimes T.T */
-  if ('url' in response.source) {
+function convertDetailsToSource(response: LayerInAreaDetails): LayerSource | null {
+  if (!('source' in response)) {
```

Comment:
Can we use typeguards?

Something like this:

```ts
type SourceContainer = GeoJSONSourceSourceContainer | TileSourceContainer


const isSourceContainer = (data: LayerInAreaDetails): data is SourceContainer => 'source' in data

const isTileSourceContainer = (data: LayerInAreaDetails): data is TileSourceContainer => {
  const casted = data as TileSourceContainer
  return casted?.source?.type === 'vector' || casted?.source?.type === 'raster'
}

function convertDetailsToSource(response: LayerInAreaDetails): LayerSource | null {
  if (!isSourceContainer(response)) {
    return null;
  }

  if (isTileSourceContainer(response)) {
    ...
```

Reply:
for readability? I guess we can

Reply:
I believe casting type intentionally when it can be avoided is bad.

Avoiding that using slightly different approach doesn't work at our typescript at least.
Therefore i wouldn't do that

![image](https://user-images.githubusercontent.com/50058726/220342276-01ad2430-4d4c-41ee-8f3f-b396c643cb1f.png)